### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by nesrv**
Syntax problems with passing parameters to the Path element

### CODE REVIEW

**Comment by nesrv** [comment]
TypeError: __init__() got an unexpected keyword argument 'd'

File: bezierDrawing.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class ExtendedHatchFill(inkex.EffectExtension):
+    """EffectExtension to fill selected objects red"""
+
+    def add_arguments(self, pars):
+        pars.add_argument("--even_color", type=inkex.Color, default=inkex.Color("blue"))
+
+    def effect(self):
+        s = self.svg
+        a = [['M ', [100, 100]],
```
Comment (toxa23) [inline bezierDrawing.py]: must be single letters, without spaces

File: bezierDrawing.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+             [' L ', [200, 300]],
+             [' L ', [600, 300]],
+             [' L ', [800, 550]],
+             [' L ', [1000, 300]],
+             [' Z', []], ]
+
+        bz = Path(d=a)  # or  bz = Path(d = f'{a}')  or bz = Path(f'{a}')
```
Comment (toxa23) [inline bezierDrawing.py]: bz = PathElement.new(a)
As you can see from Inkscape sources "Path" is a subclass of "list". In order to add it to the document we have to use ShapeElement descendants.

File: bezierDrawing.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import inkex
+from inkex import Rectangle, Transform
+from inkex import Vector2d
+from inkex import Circle, Ellipse, Polygon, Line, Rectangle, Path
+
+
+class ExtendedHatchFill(inkex.EffectExtension):
+    """EffectExtension to fill selected objects red"""
```
Comment (toxa23) [inline bezierDrawing.py]: still filling selected objects in red?

**Review by toxa23 — CHANGES_REQUESTED** [review]

File: bezierDrawing.py (hunk 2 above)
Comment (toxa23) [reply bezierDrawing.py]: or 
```
p = Path(a)
bz = PathElement.new(p)
```
in case you need the path for further modifications.

Comment (nesrv) [reply bezierDrawing.py]: @[toxa23](https://github.com/toxa23) thanks, it worked
```s = self.svg
        a = [['M', [200, 300]],
            ['Q', [400, 50, 600, 300]],
            ['T', [1000, 300]],
            ['Z', []],]
        bz = PathElement.new(a)
        bz.style = { "fill": "none", "stroke": "black" }

Comment (nesrv) [reply bezierDrawing.py]: Or is it better like this

```
def effect(self):
        s = self.svg
        tr = PathElement.new("M100 100 300 100 200 300 z")  
        # is equivalent to "M 100 100 L 300 100 L 200 300 z"
        tr.style = { "fill": "red", "stroke": "black" }
        s.add(tr)

Comment (nesrv) [reply bezierDrawing.py]: Pycharm rejects my commits, so I will write here.

```
    def effect(self):
        s = self.svg
        ds = 30
        pic = ImageGrab.grab()
        #hatching bezier selected rectangular objects
        for elem in self.svg.selection:
            b = elem.bounding_box()
            dxx = b.left # object offsets for bitmap hatching
            dyy = b.top
            for y in range(int(b.top + ds/2), int(b.top + b.height- ds/2), int(ds / 2)):
                for x in range(int(b.left), int(b.left + b.width - ds / 2 ), ds):
                    dy = randint(10,30)
                    dx = randint(30,50)
                    a = [['M', [x, y]],
                        ['Q', [x + dx, y + dy, x + 40, y]],
                        ['T', [x + 40, y]]]

                    # below is code for bitmap hatching objects (in the example by red)
                    # without debugger I can't understand why it gives me an error
                    # The error is shown at line 23 (for ...)

                    # r, g, b = pic.getpixel((int(x), int(y)))
                    # if r == 255 and g == 0 and b == 0:
                    #     hatch = Ellipse(cx=f'{x - dxx}', cy=f'{y - dyy}', rx=f'{10}', ry=f'{10}')
                    #     hatch.set_id(f'hatch' + '{x}' + '{y}')
                    #     hatch.style = { "fill": "green", "stroke": "black" }
                    #     s.add(hatch)

                    bz = PathElement.new(a)
                    bz.style = { "fill": "none", "stroke": "black"}
                    s.add(bz)

Comment (nesrv) [reply bezierDrawing.py]: image

Comment (nesrv) [reply bezierDrawing.py]: oops ... b is the color (from RGB) and b is the bounding_box

**Comment by toxa23** [comment]
do we need bezierDrawing.py in the repo? Looks like it's an experiment which results are used in ExtendedHatchFill.py

**Comment by nesrv** [comment]
> do we need bezierDrawing.py in the repo? Looks like it's an experiment which results are used in ExtendedHatchFill.py 

How do I remove it?

**Comment by nesrv** [comment]
> do we need bezierDrawing.py in the repo? Looks like it's an experiment which results are used in ExtendedHatchFill.py How do I remove it?

https://github.com/BanzaiTokyo/banzai_hatch/pull/2

**Comment by toxa23** [comment]
> > do we need bezierDrawing.py in the repo? Looks like it's an experiment which results are used in ExtendedHatchFill.py
> 
> How do I remove it?

Just delete the file.

**Comment by nesrv** [comment]
> В HTML-теге `<option value="hatch_cross">Cross</option>`«Cross» — это текст, отображаемый на экране, а «hatch_cross» — фактическое значение атрибута. Так `if self.options.fill_type == "hatch_cross"`работает отлично.

**Comment by nesrv** [comment]
> > В HTML-теге `<option value="hatch_cross">Cross</option>`«Cross» — это текст, отображаемый на экране, а «hatch_cross» — фактическое значение атрибута. Так `if self.options.fill_type == "hatch_cross"`работает отлично.

image
How do you undo it?

**Comment by toxa23** [comment]
https://github.com/rough-stuff/rough
@BanzaiTokyo , @nesrv , is this what we are trying to build?

**Comment by BanzaiTokyo** [comment]
Yes. From the first glance looks like an excellent example of what we are trying to build. 
If we can copy hatching with whole lines there, we can move further inventing our own patterns.

**Comment by nesrv** [comment]
Thanks!
I'm working on this in native python.
I also noticed the qt library 
https://doc.qt.io/qt-5/qpainterpath.html#toSubpathPolygons
which transforms bezier to broken

**Comment by nesrv** [comment]
image


Anton is Power! Respect!

**Comment by nesrv** [comment]
image

small victory!

  ```x1,y1 = pt1[0],pt1[1]
                    x2,y2 = pt2[0],pt2[1] 
                    dx = randint(-3,3)
                    dy = randint(-3,3)
                    dx1 = randint(-3,3)
                    dy2 = randint(-3,3)
                    if direction:                                        
                        path += f'M {x1}, {y1} C {x1 + dx},  {y1 + dy}  {x2 - dx1},  {y2 + dy2 } {x2}, {y2}'                   
                    else:
                        # Or go this direction                      
                        path += f'M {x2}, {y2} C {x2 + dx},  {y1 + dy}  {x1 + dx1},  {y1 +  dy2} {x1}, {y1}'
                      
                    direction = not direction

**Comment by nesrv** [comment]
image
image

**Comment by nesrv** [comment]
```
                    x1,y1 = pt1[0],pt1[1]
                    x2,y2 = pt2[0],pt2[1]
                    
                    # absolute length of the segment between the points
                    totLenLine = ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 0.5 + 0.01 # added 0.01 so it wasn't zero 
                  
                    linePX = 8 #stroke length for example
                    countLine = totLenLine / linePX # number of hatch lines
                    # coefficients of the lengths of the projections of the segments on the axes
                    dx = (x2 - x1) / totLenLine * linePX
                    dy = (y2 - y1) / totLenLine * linePX

                    if direction:
                        #Go this direction
                        for _ in range (int(countLine)) :
                             dxx = randint(-1,4) # random effect
                             dyy = randint(-2,2)
                             path += f'M {x1}, {y1} l {dx - dxx } {dy - dyy}'
                             # path += f'M {x1}, {y1} c { dxx } , {dyy} { -2* dxx},  {-2 *dyy } {dx-1 } {dy-1}'  #for bezier hatch
                             x1 += dx 
                             y1 += dy                        
                    else:
                        # Or go this direction
                        for _ in range (int(countLine)): 
                             dxx = randint(-1,4)
                             dyy = randint(-2,2)
                             path += f'M {x2 + linePX}, {y2} l {dx - dxx } {dy - dyy}' #! 0 degree angle
                             # path += f'M {x2 + linePX}, {y2 - linePX} l {dx - dxx } {dy - dyy}' # for hatching at a 45-degree angle
                             # path += f'M {x2 + linePX}, {y2} c { -dxx } , {-dyy} { 2* dxx},  {2 *dyy } { dx-1 } {dy-1}' #for bezier hatch
                             x2 -= dx 
                             y2 -= dy

**Comment by nesrv** [comment]
image

**Pull request closed without merging**